apps/openmw/mwrender/sky.cpp

```diff
@@ -557,27 +558,33 @@ class Sun : public CelestialBody
     {
         osg::ref_ptr<osg::OcclusionQueryNode> oqn = new osg::OcclusionQueryNode;
         oqn->setQueriesEnabled(true);
-
+#if OSG_VERSION_LESS_THAN(3,6,5)
+        osg::Geometry* queryGeom = oqn->getQueryGeometry();
+#else
+        osg::ref_ptr<osg::QueryGeometry> queryGeom(new osg::QueryGeometry());
+#endif
         // Make it fast! A DYNAMIC query geometry means we can't break frame until the flare is rendered (which is rendered after all the other geometry,
         // so that would be pretty bad). STATIC should be safe, since our node's local bounds are static, thus computeBounds() which modifies the queryGeometry
         // is only called once.
         // Note the debug geometry setDebugDisplay(true) is always DYNAMIC and that can't be changed, not a big deal.
-        oqn->getQueryGeometry()->setDataVariance(osg::Object::STATIC);
+        queryGeom->setDataVariance(osg::Object::STATIC);
 
         // Set up the query geometry to match the actual sun's rendering shape. osg::OcclusionQueryNode wasn't originally intended to allow this,
         // normally it would automatically adjust the query geometry to match the sub graph's bounding box. The below hack is needed to
         // circumvent this.
-        osg::Geometry* queryGeom = oqn->getQueryGeometry();
         queryGeom->setVertexArray(mGeom->getVertexArray());
         queryGeom->setTexCoordArray(0, mGeom->getTexCoordArray(0), osg::Array::BIND_PER_VERTEX);
-        queryGeom->removePrimitiveSet(0, oqn->getQueryGeometry()->getNumPrimitiveSets());
+        queryGeom->removePrimitiveSet(0, queryGeom->getNumPrimitiveSets());
         queryGeom->addPrimitiveSet(mGeom->getPrimitiveSet(0));
 
+#if OSG_VERSION_LESS_THAN(3,6,5)
         // Hack to disable unwanted awful code inside OcclusionQueryNode::computeBound.
         oqn->setComputeBoundingSphereCallback(new DummyComputeBoundCallback);
         // Still need a proper bounding sphere.
         oqn->setInitialBound(queryGeom->getBound());
```

Comment:
Are these definitely safe to remove?

Reply:
yes it's just a hack for previous design

Reply:
....but it worth the check...(even with 3.4 i don't think the initialbound is required as there's a computeboundingshperecallback...it's seams redondant..but perhaps there's stuff in openmw that mandates this initialbound...)

Reply:
I just get back to what the code...sorry for previous noise
removing these line is more than safe, it gives better result: glow don't take few frame to display when sun enter frustum